wetterdienst/file_path_handling/path_handling.py

```diff
@@ -117,6 +98,35 @@ def build_dwd_cdc_data_path(
     return f"{HTTPS_EXPRESSION}{dwd_cdc_data_path}"
 
 
+def strip_dwd_cdc_data_url_prefix(url: str, cdc_base: DWDCDCBase) -> str:
+    """
+    Strip base url away again.
+
+    [amotl] I know this looks like nonsense, but it makes the core
+            function "list_remote_files()" way more generic.
+
+    [amotl] Saying this, I want to question the fact that "cdc_base: DWDCDCBase"
+            is really passed down to these lowlevel functions. I believe this
+            should be sorted out on a higher level. Maybe we should even get rid
+            of this "fragmentation" introduced by adding radolan support through
+            05bb1c22 altogether again.
+
+            I also believe another reason for this mess was the inappropriate use of
+            "PurePosixPath()" here. It is actually URLs, so we should really use
+            "urljoin()" all over the place.
+
+    [amotl] When reverting the fragmentation introduced recently by 05bb1c22,
+            this function would not have been needed. There you go.
+
+    :param url:
+    :param cdc_base:
+    :return:
+    """
+    dwd_cdc_data_prefix = f"{HTTPS_EXPRESSION}{str(PurePosixPath(DWD_SERVER, DWD_CDC_PATH, cdc_base.value))}/"
+    url = url.replace(dwd_cdc_data_prefix, "")
+    return url
+
+
```

Comment:
Dear @gutzbenj,

we can remove this whole function again if you would consider reverting the "fragmentation" of the signatures of these url accessor functions into _path_ vs. _cdc_base_, coming from 05bb1c22 just recently. I believe it was way better before.

Maybe we are able to still use the idea of `DWDCDCBase` on the calling site?

With kind regards,
Andreas.

P.S.: Sorry for my ~rant~ lengthy explanation. This workaround has not been part of the original pull request and has taken a significant time to figure out because everything has been so generic on this level before. Now, we need this function to compensate.

Reply:
I get what you mean! I will revert the fragmentation and instead use the base url, like you said, on the calling side

Reply:
Thank you so much for #183, @gutzbenj!

And at the same time, sorry about the disruptions this might be causing for #140, @meteoDaniel! I recognized you started to build upon many things how the file index building and downloading worked before. However, I believe the core implementation is way more generic and straightforward now, so I believe it will be worth the effort of adjusting accordingly.